Automattic/jetpack
# Connection: handle connection errors in the package state

#### Changes proposed in this Pull Request:
Add the connection verified errors to the @wordpress/data. This is part of a bigger project (1202697549597858-as) to provide consistent failed connection error reporting and an interface to reconnect (1202697549597858-as-1202697549597864). 

The changes will affect the following:
* Jetpack
* Connection

#### Other information:


#### Jetpack product discussion
* p9dueE-5w8-p2

#### Does this pull request change what data or activity we track or use?
* No

#### Testing instructions:
* Run `jetpack build plugins/jetpack`.
* Go to Jetpack > Dashboard.
* Open up the browser's developer console.
* Run the following console commands:
```
wp.data.select('jetpack-connection').getConnectionErrors()
errors = ['error1', 'error2'];
wp.data.dispatch('jetpack-connection').setConnectionErrors(errors)
wp.data.select('jetpack-connection').getConnectionErrors()
```
You should see output similar to the following (example output from Google Chrome):
```
> wp.data.select('jetpack-connection').getConnectionErrors()
{}
> errors = ['error1', 'error2'];
(2) ['error1', 'error2']
> wp.data.dispatch('jetpack-connection').setConnectionErrors(errors)
Promise {<fulfilled>: {…}}
> wp.data.select('jetpack-connection').getConnectionErrors()
(2) ['error1', 'error2']
```
* Verify that errors can be set with `setConnectionErrors()` and the set errors are returned with `getConnectionErrors()` as described above.
